text/0000-if-let-chains.md

```diff
@@ -0,0 +1,2003 @@
+    > In fact, presenting alternatives to Rust's existing `&&` syntax for
+    > ANDing together terms made even the use of `&&` confusing because the
+    > claim that they were all equivalent meant that it "couldn't possibly be"
+    > the existing meaning of `&&`. I didn't know what was going on until I
+    > realized I'd glossed over tiny (ie. unimportant) text which actually
+    > explained the meaning in plain English... at which point, I realized that
+    > the syntaxes other than `&&` had set up a mistaken assumption that ruled
+    > out the actual proper interpretation.
+
+45. > I like Using `&&` for conjunction and `let PATTERN = EXPR`  because it,
+    > for me, has the least surprises syntactically. `&&` indicates conjunction
+    > of the predicate, and including `x` and `y` in subsequent scopes is
+    > something I wish existed, but if we're not clear about it, it could get messy. 
+
+46. > it does not introduce anything fancy new stuff
+
+47. > I like the "`let`" syntax better than the `EXPR is PATTERN` syntax,
+    > since it's used in other places already.
+
+48. > 1. `&&` is already a familiar concept for working with boolean expressions
+    > 2. `if let` is how we already achieve conditional binding
+    >
+    > The combination of "`is`" and `&&` is the only other choice I could
+    > consider, albeit begrudgingly. I'm kind of uncomfortable with giving
+    > up keyword real estate and having another way of doing `if let`.
+    >
+    > Other than that, I feel like the other choices alienate both new and old
+    > Rust programmers alike. We should be focusing on keeping things as simple
+    > and familiar as possible.
+
+## Appendix B.3
+[Appendix B.3]: #appendix-b3
+
+Some survey participants did not explicitly say that the RFC's proposed syntax
+was *consistent*, but they did say, in some way, that it was *intuitive*.
+
+1. > `&&` is the logical conjunction operator and `let A(x) = foo` clearly
+   > destructures for pattern matching
+
+2. > The most intuitive
+
+3. > It is not surprising
+
+4. > Looks like straightforward boolean logic, the rest seem like arcane syntax.
+
+5. > Reminiscent of boolean algebra
+
+6. > It fits with my mental model of how patterns and Boolean logic work
+
+7. > `&&` makes the logic relationship clearer, and using `let` for binding is the same. 
+   > Conjunction is more readable with `&&`
+
+8. > We already have “`if let`” elsewhere. Don’t introduce a new “`is`” syntax
+   > here, it’s not any more intuitive.
+
+## Appendix B.4
+[Appendix B.4]: #appendix-b4
+
+Some survey participants felt that the proposed syntax set up the expectation
+of `let PAT = EXPR` being an expression typed at `bool` as opposed to a
+statement which is currently the case.
+
+1. > `&&` as separator would require boolean expressions.
+
+2. > `&&` is for boolean expressions, and won't work right in generic usage.
+   > `let` would have to return a boolean which is weird and probably a breaking change.
```

Comment:
Again something that would become obvious if `&& if let` were used instead of `&& let`

```rust
let x: bool = if let a...
```

Already exists today and needs a bool expression in the body and an else branch

Reply:
Associated discussion: https://github.com/Centril/rfcs/blob/rfc/let-chains-2/text/0000-if-let-chains.md#expectation-that-let-pat--expr--bool

TL;DR: this expectation can be readily ruled out with (good) error messages and `if let` already induces this problem by making users think that it is the composition of `if` + `let PAT = EXPR`. I don't think it will be that big of a problem in practice.